Framework/Algorithms/src/ConvertUnits.cpp

```diff
@@ -80,19 +74,58 @@ void ConvertUnits::init() {
                   "have identical bin boundaries. This option is not "
                   "recommended (see "
                   "http://www.mantidproject.org/ConvertUnits).");
+
+  declareProperty(
+      "ConvertFromPointData", true,
+      "When checked, if the Input Workspace contains Points\n"
+      "the algorithm ConvertToHistogram will be run to convert\n"
+      "the Points to Bins. The Output Workspace will contains Bins.");
 }
 
 /** Executes the algorithm
- *  @throw std::runtime_error If the input workspace has not had its unit set
- *  @throw NotImplementedError If the input workspace contains point (not
- * histogram) data
- *  @throw InstrumentDefinitionError If unable to calculate source-sample
- * distance
- */
+*  @throw std::runtime_error If the input workspace has not had its unit set
+*  @throw NotImplementedError If the input workspace contains point (not
+* histogram) data
+*  @throw InstrumentDefinitionError If unable to calculate source-sample
+* distance
+*/
 void ConvertUnits::exec() {
   // Get the workspaces
+  bool acceptPointData = getProperty("ConvertFromPointData");
+
   MatrixWorkspace_sptr inputWS = getProperty("InputWorkspace");
-  this->setupMemberVariables(inputWS);
+
+  // Holder for the correctWS, because if we're converting from
+  // PointData a new workspace is created
+  MatrixWorkspace_sptr correctWS;
+  if (!inputWS->isHistogramData()) {
+
+    if (acceptPointData) {
+      g_log.information(
+          "ConvertFromPointData is checked. Running ConvertToHistogram\n");
+      // not histogram data
+      // ConvertToHistogram
+      IAlgorithm_sptr convToHist = createChildAlgorithm("ConvertToHistogram");
+      convToHist->setProperty("InputWorkspace", inputWS);
+      convToHist->execute();
+      MatrixWorkspace_sptr temp = convToHist->getProperty("OutputWorkspace");
+      correctWS = boost::dynamic_pointer_cast<MatrixWorkspace>(temp);
+
+      if (!correctWS->isHistogramData()) {
+        throw std::runtime_error(
+            "Failed to convert workspace from Points to Bins");
+      }
+    } else {
+      throw std::runtime_error("Workspace contains points, you can either run "
+                               "ConvertToHistogram on it, or check "
```

Comment:
"or check" implies to me the user needs to actually check something (i.e. verify something). Could we say "or set ConvertFromPointData to enabled"